libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -0,0 +1,2289 @@
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+import android.text.TextUtils;
+import android.text.TextWatcher;
+import android.text.style.ForegroundColorSpan;
+import android.text.style.SuggestionSpan;
+import android.view.ContextThemeWrapper;
+import android.view.DragEvent;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.BaseInputConnection;
+import android.webkit.URLUtil;
+import android.widget.Toast;
+
+
+import com.android.volley.toolbox.ImageLoader;
+import com.google.gson.Gson;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wordpress.android.editor.MetadataUtils.AttributesWithClass;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+import org.wordpress.android.util.DisplayUtils;
+import org.wordpress.android.util.ProfilingUtils;
+import org.wordpress.android.util.StringUtils;
+import org.wordpress.android.util.ToastUtils;
+import org.wordpress.android.util.UrlUtils;
+import org.wordpress.android.util.helpers.MediaFile;
+import org.wordpress.android.util.helpers.MediaGallery;
+import org.wordpress.aztec.AztecAttributes;
+import org.wordpress.aztec.AztecExceptionHandler;
+import org.wordpress.aztec.AztecParser;
+import org.wordpress.aztec.AztecText;
+import org.wordpress.aztec.Html;
+import org.wordpress.aztec.IHistoryListener;
+import org.wordpress.aztec.extensions.MediaLinkExtensionsKt;
+import org.wordpress.aztec.plugins.IAztecPlugin;
+import org.wordpress.aztec.plugins.IMediaToolbarButton;
+import org.wordpress.aztec.plugins.shortcodes.AudioShortcodePlugin;
+import org.wordpress.aztec.plugins.shortcodes.CaptionShortcodePlugin;
+import org.wordpress.aztec.plugins.shortcodes.VideoShortcodePlugin;
+import org.wordpress.aztec.plugins.shortcodes.extensions.CaptionExtensionsKt;
+import org.wordpress.aztec.plugins.shortcodes.extensions.VideoPressExtensionsKt;
+import org.wordpress.aztec.source.Format;
+import org.wordpress.aztec.source.SourceViewEditText;
+import org.wordpress.aztec.spans.IAztecAttributedSpan;
+import org.wordpress.aztec.toolbar.AztecToolbar;
+import org.wordpress.aztec.util.AztecLog;
+import org.wordpress.aztec.watchers.EndOfBufferMarkerAdder;
+import org.wordpress.mobile.WPAndroidGlue.WPAndroidGlueCode;
+import org.wordpress.mobile.WPAndroidGlue.WPAndroidGlueCode.OnGetContentTimeout;
+import org.wordpress.mobile.WPAndroidGlue.WPAndroidGlueCode.OnMediaLibraryButtonListener;
+import org.xml.sax.Attributes;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+public class GutenbergEditorFragment extends EditorFragmentAbstract implements
```

Comment:
I like the idea of moving to kotlins and VMs as soon as possible, especially since apparently we're already adding features from scratch to Gutenberg's case. Yes, not something to tackle in this PR but defo a good idea to pursue soon!

Reply:
Great to hear! As we also talked on the help, I am happy to help in any way I can with this.